### PR TITLE
Don't invalidate SuperView if not connected to Window

### LINKED ISSUE #24032: Crash on ios when open new screen or navigate back to it
### Description

It seems, there were similar issues previously
https://github.com/dotnet/maui/issues/23139, 
but now issue appeared after upgrade to maui to 8.0.70 and removing .UseMauiCompatibility()
Sometimes view try to created with NativeHandle argument in constructor

This crash can appeared with Microsoft.Maui.Platform.ContentView|LayoutView and some custom views(so, adding constructor with ObjCRuntime.NativeHandle resolve issue for custom views, but no any options to resolve crashes with ContentView, LayoutView)

### Steps to Reproduce

Sometimes just randomly appeared when navigate to new screen or navigate back to existing

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.40 SR5

### Affected platforms

iOS

### Affected platform versions

_No response_

### Did you find any workaround?

For custom views we can just add constructor with ObjCRuntime.NativeHandle handle, but no any ways to fix ios ContentView/LayoutView

### Relevant log output

```shell
Unhandled managed exception: Failed to marshal the Objective-C object 0x14177b170 (type: Microsoft_Maui_Platform_ContentView). Could not find an existing managed instance for this object, nor was it possible to create a new managed instance (because the type 'Microsoft.Maui.Platform.ContentView' does not have a constructor that takes one NativeHandle argument). (ObjCRuntime.RuntimeException)
   at ObjCRuntime.Runtime.MissingCtor(IntPtr , IntPtr , Type , MissingCtorResolution , IntPtr , RuntimeMethodHandle )
   at ObjCRuntime.Runtime.ConstructNSObject[UIView](IntPtr , Type , MissingCtorResolution , IntPtr , RuntimeMethodHandle )
   at ObjCRuntime.Runtime.GetNSObject[UIView](IntPtr , IntPtr , RuntimeMethodHandle , Boolean )
   at ObjCRuntime.Runtime.GetNSObject[UIView](IntPtr , IntPtr , RuntimeMethodHandle )
   at ObjCRuntime.Runtime.GetNSObject[UIView](IntPtr )
   at UIKit.UIView.get_Superview()
   at Microsoft.Maui.Platform.MauiView.SetNeedsLayout()
```

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

I rolled back these changes on a previous PR because they caused a regression when modifying a layout that's offscreen
https://github.com/dotnet/maui/pull/24477

From testing, I found that the main part that caused a regression was the "setNeedsLayout" changes inside LayoutView. 

It's interesting because now with this [PR](https://github.com/dotnet/maui/pull/23052) the setNeedsLayout calls inside LayoutView aren't really even needed. 

A user reported that they were occasionally still seeing the exception so I've taken a different approach here and have it refire `SetNeedsLayout` on the parent if the view gets reattached to the window.  

### Issues Fixed

Fixes #24032

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue24434.cs [hunk 1]
```diff
@@ -24,17 +24,17 @@ public TestPage()
...
-							taskCompletionSource.TrySetResult();
-						};
+						var secondPage = new ContentPage() { Content = new Label() { Text = "I should just disappear" } };
 
-						await Navigation.PushAsync(new ContentPage() { Content = new Label() { Text = "I should just disappear" } });
+						await Navigation.PushAsync(secondPage);
 						await Navigation.PushModalAsync(new ContentPage() { Content = new Label() { Text = "I should just disappear" } });
 
-						await taskCompletionSource.Task.WaitAsync(TimeSpan.FromSeconds(5));
+						// Ensure that the VSL is unloaded otherwise the test isn't really valid
```
Comment (mattleibow) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue24434.cs:32]: If the VSL is not unloaded, what does that mean? The test will pass and we will not know? Should we skip/fail the test if the VSL is not unloaded? Can we trigger a GC to force something?

TL;DR: this code has a special path for unloaded VSL, but what if it is loaded?

Comment (PureWeen) [reply src/Controls/tests/TestCases.HostApp/Issues/Issue24434.cs:32]: If the VSL is not unloaded its still part of the visual tree. This isn't really related to being garbage collected. 

What we're validating here is that if I add a view, while the VSL isn't part of the visual tree, once it gets added back to the visual tree it will be visible.

The test side of this is checking for the existence of the label on the screen

https://github.com/dotnet/maui/blob/75580ddbd0773dc76bf2188f6b285df4b2820d16/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24434.cs#L21

So, if the label doesn't get added in code or renderer by the platform then the test will fail

**Review by mattleibow — APPROVED** [review]
